src/Core/Services/CryptoService.cs

```diff
@@ -36,189 +34,268 @@ public class CryptoService : ICryptoService
             _cryptoFunctionService = cryptoFunctionService;
         }
 
-        public async Task SetKeyAsync(SymmetricCryptoKey key)
+        public void ClearCache()
         {
-            await _stateService.SetKeyDecryptedAsync(key);
-            var option = await _stateService.GetVaultTimeoutAsync();
-            var biometric = await _stateService.GetBiometricUnlockAsync();
-            if (option.HasValue && !biometric.GetValueOrDefault())
-            {
-                // If we have a lock option set, we do not store the key
-                return;
-            }
-            await _stateService.SetKeyEncryptedAsync(key?.KeyB64);
+            _legacyEtmKey = null;
+            _masterKeyHash = null;
+            _publicKey = null;
+            _privateKey = null;
+            _orgKeys = null;
         }
 
-        public async Task SetKeyHashAsync(string keyHash)
+        public async Task ToggleKeysAsync()
         {
-            _keyHash = keyHash;
-            await _stateService.SetKeyHashAsync(keyHash);
+            // refresh or clear the pin key
+            await SetUserKeyAsync(await GetUserKeyAsync());
+
+            // refresh or clear the encrypted user key
+            var encUserKey = await _stateService.GetUserKeyMasterKeyAsync();
+            await _stateService.SetUserKeyMasterKeyAsync(null);
+            await _stateService.SetUserKeyMasterKeyAsync(encUserKey);
```

Comment:
I think I was trying to stay consistent with the old version of this method. All this does is grab the encrypted key from state, clear the state, and set it again. I'm not sure there's much value here though since there's not side effects so I'm going to remove it.

Let me know if you think it needs to stay, and good catch!